test/DebtToken.t.sol

```diff
@@ -88,79 +96,111 @@ contract DepositAndWithdrawalTest is DebtTokenTest {
     //deposit
     function testRevert_DepositUnauthorised(uint128 assets, address receiver, address unprivilegedAddress) public {
         vm.assume(unprivilegedAddress != address(pool));
+        // Given: all neccesary contracts are deployed on the setup
 
+        // When: unprivilegedAddress is pranked
         vm.startPrank(unprivilegedAddress);
+        // Then: deposit should revert with UNAUTHORIZED
         vm.expectRevert("UNAUTHORIZED");
         debt.deposit(assets, receiver);
         vm.stopPrank();
     }
 
     function testRevert_DepositZeroShares(address receiver) public {
+        // Given: all neccesary contracts are deployed on the setup
+
+
+        // When: pool is pranked
         vm.startPrank(address(pool));
```

Comment:
This is a tricky one, the When and Then with unhappy flows happen in the same line. The correct are here: 
When: depositing zero shares
Then: deposit should revert with ZERO_SHARES

I propose to do (and for all other unhappy flows):

vm.startPrank(address(pool));
// When: depositing zero shares
// Then: deposit should revert with ZERO_SHARES
vm.expectRevert("ZERO_SHARES");
debt.deposit(0, receiver);
vm.stopPrank();

But let's discuss tomorrow during standup!

Reply:
This makes much more sense, will do